lib/controllers/flex-controller.js

```diff
@@ -0,0 +1,270 @@
+        const isDomainApp = flexNs.domain === DomainTypes.APP;
+        const domainEntity = {
+          id: flexNs.domainEntityId,
+          schemaVersion: flexNs.schemaVersion
+        };
+
+        this.flexService.getServices(isDomainApp, domainEntity, (err, services) => {
+          if (err) {
+            return next(err);
+          }
+
+          this.cliManager.log(LogLevel.INFO, 'You have %s Kinvey service connectors:', chalk.cyan(services.length));
+
+          services.forEach((service) => {
+            const bullet = service.id === flexNs.serviceId ? chalk.green('* ') : '';
+            this.cliManager.log(LogLevel.INFO, '%s%s', bullet, chalk.cyan(service.name));
+          });
+
+          this.cliManager.log(LogLevel.INFO, 'The service used in this project is marked with *');
+
+          next(null, services);
+        });
+      }
+    ], (err, results) => {
+      const result = results.pop();
+      this.cliManager.processCommandResult(err, result, done);
+    });
+  }
+
+  logs(options, done) {
+    async.series([
+      (next) => {
+        super.processAuthOptions(options, next);
+      },
+      (next) => {
+        // FIXME: Stop handling them before #BACK-2775 is merged into master
+        // Handle deprecated logs command params
+        if (options._.includes(FlexOptionsNames.FROM) || options._.includes(FlexOptionsNames.TO)) {
+          return next(new KinveyError('DeprecationError', `Version 1.x ${chalk.whiteBright('[from]')} and ${chalk.whiteBright('[to]')} params have been converted to options. Use ${chalk.blueBright('--from')} and ${chalk.blueBright('--to')} to filter by timestamp instead.`));
+        }
+
+        this.flexService.validateLogsOptions(options, next);
+      },
+      (next) => {
+        const flexNs = this.projectSetup.getFlexNamespace();
+        this.flexService.getServiceLogs(options, flexNs.serviceId, flexNs.schemaVersion, (err, logs) => {
+          if (err) {
+            return next(err);
+          }
+
+          this.flexService.printServiceLogs(flexNs, logs);
+          next(null, logs);
+        });
+      },
+    ], (err, results) => {
+      const result = results.pop();
+      this.cliManager.processCommandResult(err, result, done);
+    });
+  }
+
+  recycle(options, done) {
+    async.series([
+      (next) => {
+        super.processAuthOptions(options, next);
+      },
+      (next) => {
+        const flexNs = this.projectSetup.getFlexNamespace();
+        this.flexService.execRecycle(flexNs.schemaVersion, flexNs.serviceId, (err, jobId) => {
+          if (err) {
+            return next(err);
+          }
+
+          this.cliManager.log(LogLevel.INFO, 'Recycle initiated, received job %s', chalk.cyan(jobId));
+          next(null, jobId);
+        });
+      }
+    ], (err, results) => {
+      const result = results.pop();
+      this.cliManager.processCommandResult(err, result, done);
+    });
+  }
+
+  deleteProjectSetup(done) {
+    const err = this.ensureFlexSetup();
```

Comment:
If this is synchronous you should consider using `try`/`catch` here (convention/easier to understand).

Reply:
I've considered it but then I'd have repeating logic for the error handling - try, catch and processing the error. Seems to me it's better to have a `try catch` in the function doing the sync operation and afterwards just check for error. As long as the error handling of sync operations is consistent throughout the code, I think we are fine.

Reply:
That might be harder to enforce if different people start to contribute, though, since `try/catch` is well understood but `const err = this.ensureFlexSetup()` is something that we created

Reply:
Actually, it is something that I found while thinking what would be a proper solution.

https://stackoverflow.com/questions/7310521/node-js-best-practice-exception-handling/7313005#7313005

https://gist.github.com/balupton/5560110